x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_icon_property.tsx

```diff
@@ -115,21 +117,26 @@ export class DynamicIconProperty extends DynamicStyleProperty<IconDynamicOptions
   renderLegendDetailRow({ isPointsOnly, isLinesOnly }: LegendProps) {
     const { stops, fallbackSymbolId } = this._getPaletteStops();
     const breaks = [];
+    const layerStyle = this._layer.getCurrentStyle() as IVectorStyle;
     stops.forEach(({ stop, style }) => {
       if (stop) {
+        const { svg } = layerStyle.getIconMeta(style);
         breaks.push({
           color: 'grey',
           label: this.formatField(stop),
           symbolId: style,
+          svg,
         });
       }
     });
 
     if (fallbackSymbolId) {
+      const { svg } = layerStyle.getIconMeta(fallbackSymbolId);
```

Comment:
change to `const svg = layerStyle.getIconMeta(fallbackSymbolId)?.svg;`, getIconMeta can return undefined